rules_gapic/gapic.bzl

```diff
@@ -62,9 +76,21 @@ gapic_srcjar = rule(
 def _proto_custom_library_impl(ctx):
     imports = depset()
     srcs = depset()
+    transitive_descriptor_sets = depset()
+    check_dep_sources = depset()
+
+    cur_package = ctx.label.package
     for dep in ctx.attr.deps:
-        srcs = depset(direct = srcs.to_list(), transitive = [dep.proto.check_deps_sources])
+        src = dep.proto.check_deps_sources
+        srcs = depset(direct = srcs.to_list(), transitive = [src])
         imports = depset(direct = imports.to_list(), transitive = [dep.proto.transitive_imports])
+        transitive_descriptor_sets = depset(
```

Comment:
Is this the recommended way to add to a depset? What about constructing a normal list of all of the `dep.proto.transitive_descriptor_sets` objects in the loop, and then constructing a single depset after the loop?

Reply:
I'm not sure if it is recommended, but I believe I "stole" it from some examples. Most probably it is inefficient, but on practice, since it is proto depsets, the depset "tree" and number of iterations in the list is very small (in most cases there is only one element), so it should be ok.

But in any case i think you are right. I changed it so it just adds all to the list, and then creates a depset.